incubator/hnc/internal/reconcilers/object.go

```diff
@@ -699,6 +705,26 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	}
 }
 
+// getNoneSelector returns true indicates that user do not want this object to be propagated
+func (r *ObjectReconciler) getNoneSelector(log logr.Logger, inst *unstructured.Unstructured) bool {
+	annot := inst.GetAnnotations()
+	noneSelectorStr, ok := annot[api.AnnotationNoneSelector]
+	if !ok {
+		return false
+	}
+	// Empty string is treated as 'false'
+	if noneSelectorStr == "" {
+		return false
+	}
```

Comment:
This explicit `return false` isn't part of the other two functions. If you pass an empty string as `selector` or `treeSelector`, do those count as "match anything" or "match nothing?" Given that the tests are passing, I'm guessing that we interpret an empty string as "match everything" but we should probably specify that somewhere.

Reply:
The empty string selector matches everything - the reason why I add this line here is because we are using `noneselector` as a boolean value (not a `label.selector`), so we need to specifically handle the empty string case

I've added some comments here

Reply:
sg